include/boost/math/special_functions/gamma.hpp

```diff
@@ -366,13 +366,34 @@ std::size_t highest_bernoulli_index()
 template<class T>
 int minimum_argument_for_bernoulli_recursion()
 {
-   const float digits10_of_type = (std::numeric_limits<T>::is_specialized
-                                      ? static_cast<float>(std::numeric_limits<T>::digits10)
-                                      : static_cast<float>(boost::math::tools::digits<T>() * 0.301F));
+   BOOST_CONSTEXPR_OR_CONST float digits10_of_type =
```

Comment:
Unfortunate point of pedantry: this can't be constexpr since `tools::digits<>()` can't (always) be constexpr: for example with types whose precision varies at runtime (mpfr_float for eg).

Reply:
Thanks. Correct.
I will change accordingly.